src/main/java/strategy/BalanceOperation.java

```
@@ -0,0 +1,11 @@
+package strategy;
+
+import db.Storage;
+import model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.setQuantity(transaction.getFruit(), transaction.getQuantity());
```

You call `Storage.setQuantity(transaction.getFruit(), transaction.getQuantity())` without any validation or exception context. This can lead to invalid state (negative quantities) or obscure errors later. Before mutating storage validate inputs locally (as noted above). Additionally, the requirements say: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and original cause." If Storage methods can throw, catch and rethrow with transaction context so callers see which CSV row/transaction caused the failure.